### ISSUE TITLE
[Proposal] Reduce warnings in flutter.groovy file 

### ISSUE THREAD

**Issue by AbdeMohlbi** [flutter/flutter#147122]
### Use case

right now **(using flutter 3.19.6)** when opening the _android module_ in android studio the `flutter.groovy` file located inside `...\flutter\packages\flutter_tools\gradle\src\main\groovy\flutter.groovy` needs some refactoring and cleaning because it has more then 70 warnings right now (some libraries are deprecated,some unused methods ...): 
i couldn't upload the code cz github has a limit in the body length so it's [here](https://github.com/AbdeMohlbi/flutter_groovy_pr/blob/main/flutter_old.groovy)

### Proposal

the updated version can be found [here](https://github.com/AbdeMohlbi/flutter_groovy_pr/blob/main/flutter_new.groovy)
note that i haven't change or delete any function or messed any kind of logic just cleaned a bit and did as same as [here](https://github.com/flutter/flutter/issues/146036) to get rid of buildDir deprecation(there is now 31 warnings )

**Comment by AbdeMohlbi** [comment]
also something strange is here (that ididn't touch )
```groovy
private String resolveProperty(String name, String defaultValue) {
        if (localProperties == null) {
            localProperties = readPropertiesIfExist(new File(project.projectDir.parentFile, "local.properties"))
        }
        String result
        if (project.hasProperty(name)) {
            result = project.property(name)
        }
        if (result == null) {
            result = localProperties.getProperty(name)
        }
        if (result == null) {
            result = defaultValue
        }
        return result
    }
```

**Comment by AbdeMohlbi** [comment]
also i have seen what the file in master branch and it had most of the fixes and i tried to make a combination of the two by using my fixes inside this version  [here is the final suggestion](https://github.com/AbdeMohlbi/flutter_groovy_pr/blob/master/flutter_new.groovy)

**Comment by huycozy** [comment]
Thanks for the proposal. This is number of warnings I see when opening that file on Android Studio:

- Flutter stable `3.19.6`: 61 warnings
- Flutter master channel `3.22.0-14.0.pre.67`: 24 warnings

**Comment by AbdeMohlbi** [comment]
@huycozy  
61 using the stable release flutter.groovy file not mine u mean ? 
Well try my code inside the repo and see 
Gow many u get 
Also at most this afternon i will tag u another time to test the flutter.groovy file for the master branch because it seems that u have same master version as me

**Comment by huycozy** [comment]
@AbdeMohlbi Yes, I checked this (warnings in file `flutter.groovy`) with official Flutter stable and master releases.
